subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/TaskPropertyInfo.java

```diff
@@ -103,20 +104,20 @@ public Object create() {
         return new TaskPropertyValue() {
             @Override
             public Object getValue() {
-                return value;
+                return DeferredUtil.unpack(value);
```

Comment:
@big-guy I decided to unpack it here instead. Do you think it's reasonable? The other alternative is to unpack only during the check but I fear the consumer of `getValue()` may also want to unpacked value.